packages/cli-repl/src/arg-mapper.ts

```diff
@@ -103,6 +104,25 @@ export function applyTlsCertificateSelector(
   }
 }
 
+/**
+   * Applies driverInfo to track usage of mongosh,
+   * because appName can be overwritten by a user in their connection string.
+   *
+   * @param {MongoClientOptions} nodeOptions - The MongoClient options.
+   *
+   * @returns {DriverInfo} The driverInfo metadata.
```

Comment:
I think it would be a bit cleaner if the PR followed this line, i.e. instead of modifying a passed-in object of options, we only return the `DriverInfo` as an object

Reply:
I aligned the behavior with the applyTlsCertificateSelector: https://github.com/mongodb-js/mongosh/pull/1190/files#diff-44d5c1b1de68d646516652eae9e4eaab41715e8afb81936829928aa1135b5e75R72-R73

I get that applyTlsCertificateSelector does more stuff, but the use cases feel very similar to me.

Reply:
That’s a good point – I think we should also be doing this differently in `applyTlsCertificateSelector` :slightly_smiling_face:  If it’s alright with you, you can fix that, or let me do that after this PR is merged (it could return `MongoClientOptions` instead of taking it as an argument).

Reply:
Sure I will fix it! Just wanted to make sure I don't miss any reason why they might be different.